### ISSUE TITLE
link helper should accept function (or block) for embedded content

### ISSUE THREAD

**Issue by chrismccord** [phoenixframework/phoenix_html#2]
ie:

```erb
<%= link to: user_path(@conn, :show, user), fn -> %>
  <h3><%= user.handle %></h3>
<% end %>

```

// @josevalim

**Comment by josevalim** [comment]
:+1:. It should probably be though:

```eex
<%= link to: user_path(@conn, :show, user) do %>
  <h3><%= user.handle %></h3>
<% end %>
```

As that's what we use for content tags.

**Comment by chrismccord** [comment]
That works!

**Comment by whatyouhide** [comment]
I think this can be closed after #4 has been merged!

**Comment by chrismccord** [comment]
Thanks!
